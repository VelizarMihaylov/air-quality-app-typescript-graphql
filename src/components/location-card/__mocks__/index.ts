export const measurements = [
  {
    __typename: 'Measurement',
    parameter: 'o3',
    value: 25,
    unit: 'µg/m³'
  },
  {
    __typename: 'Measurement',
    parameter: 'so2',
    value: 12,
    unit: 'µg/m³'
  },
  {
    __typename: 'Measurement',
    parameter: 'pm25',
    value: 6,
    unit: 'µg/m³'
  },
  {
    __typename: 'Measurement',
    parameter: 'no2',
    value: 16,
    unit: 'µg/m³'
  }
]
