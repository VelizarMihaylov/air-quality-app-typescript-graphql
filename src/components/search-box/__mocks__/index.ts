export const cities = [
  {
    name: 'Manchester'
  },
  {
    name: 'London'
  },
  {
    name: 'Liverpool'
  },
  {
    name: 'Birmingham'
  }
]
