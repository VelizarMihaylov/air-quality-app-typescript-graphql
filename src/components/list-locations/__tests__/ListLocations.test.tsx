import React from 'react'
import { ListLocations } from '..'
import { create, act } from 'react-test-renderer'

import { ThemeProvider } from 'styled-components'
import generateTheme from 'theme'

import { locations } from '../__mocks__'

import dayjs from 'dayjs'
jest.mock('dayjs')

const mockedDayJs: jest.Mock<unknown> = dayjs as never

describe('ListLocations', () => {
  it('should handle loading state', () => {
    const ListLocationsRender = create(
      <ThemeProvider theme={generateTheme()}>
        <ListLocations city="Manchester" />
      </ThemeProvider>
    ).toJSON()
    expect(ListLocationsRender).toMatchSnapshot()
  })
  it('should handle error state', () => {
    const ListLocationsRender = create(
      <ThemeProvider theme={generateTheme()}>
        <ListLocations city="Manchester" error />
      </ThemeProvider>
    ).toJSON()
    expect(ListLocationsRender).toMatchSnapshot()
  })

  it('should render locations when not in loading or error state and passed latestMeasurements', () => {
    mockedDayJs.mockImplementation(() => ({
      extend: jest.fn(),
      utc: (): { fromNow: () => string } => ({
        fromNow: (): string => '5 hours ago'
      })
    }))
    const { latestMeasurements } = locations
    const ListLocationsRender = create(
      <ThemeProvider theme={generateTheme()}>
        <ListLocations city="Manchester" locations={latestMeasurements} />
      </ThemeProvider>
    ).toJSON()
    expect(ListLocationsRender).toMatchSnapshot()
  })
})
