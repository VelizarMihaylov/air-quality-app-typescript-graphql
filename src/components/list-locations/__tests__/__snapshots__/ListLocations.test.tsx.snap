// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListLocations should handle error state 1`] = `
<h1>
  Oops something went wrong!
</h1>
`;

exports[`ListLocations should handle loading state 1`] = `
<div
  className="sc-jSgupP cliOlb"
/>
`;

exports[`ListLocations should render locations when not in loading or error state and passed latestMeasurements 1`] = `
<div
  className="sc-jSgupP cliOlb"
>
  <div
    className="sc-hKgILt xsnPe location-card"
    data-puppet="location-card"
  >
    <div
      className="card-content"
    >
      <p
        className="updated-at"
      >
        Updated 5 hours ago
      </p>
      <h3>
        Manchester Piccadilly
      </h3>
      <p>
        in Manchester, United Kingdom
      </p>
      <p
        className="values"
      >
        Value: PM25: 10 µg/m³, SO2: 11 µg/m³, NO2: 22 µg/m³, O3: 17 µg/m³
      </p>
    </div>
  </div>
  <div
    className="sc-hKgILt xsnPe location-card"
    data-puppet="location-card"
  >
    <div
      className="card-content"
    >
      <p
        className="updated-at"
      >
        Updated 5 hours ago
      </p>
      <h3>
        Manchester Sharston
      </h3>
      <p>
        in Manchester, United Kingdom
      </p>
      <p
        className="values"
      >
        Value: O3: 34 µg/m³, NO2: 10 µg/m³
      </p>
    </div>
  </div>
  <div
    className="sc-hKgILt xsnPe location-card"
    data-puppet="location-card"
  >
    <div
      className="card-content"
    >
      <p
        className="updated-at"
      >
        Updated 5 hours ago
      </p>
      <h3>
        Salford Eccles
      </h3>
      <p>
        in Manchester, United Kingdom
      </p>
      <p
        className="values"
      >
        Value: NO2: 13 µg/m³, PM25: 13 µg/m³, PM10: 16 µg/m³
      </p>
    </div>
  </div>
</div>
`;
